### ISSUE TITLE
Duplicated foreign keys and indexes creation during server deployment

### ISSUE THREAD

**Issue by francielelithg** [strapi/strapi#16631]
**Node version:** v16.19.1
**Strapi version:** v4.10.2
Using MySQL

Hey guys,

Recently I have set up a new server for the latest Strapi version and everything has been working properly during implementation on my local, but once I pushed it to staging environment (separate database per envs), I am facing this issue:

`[2023-05-04 01:41:58.170] error: alter table `upload_folders_parent_links` add constraint `upload_folders_parent_links_fk` foreign key (`folder_id`) references `upload_folders` (`id`) on delete CASCADE - ER_FK_DUP_NAME: Duplicate foreign key constraint name 'strapi_v4/upload_folders_parent_links_fk'`

This is happening for different indexes and/or foreign keys (every deployment it indicates an error on a different table, but all of them, ending with '_links').

Any clue on what may be happening? I've been searching through the community but could not find similar issues. #11421 and #13401 are similar but don't provide a fix for it.

**Comment by Marc-Roig** [comment]
hello :) Did your staging environment was running on an older version before pushing ?If so , could you share the version it was using before?

**Comment by francielelithg** [comment]
> hello :) Did your staging environment was running on an older version before pushing ?If so , could you share the version it was using before?

Hi! I was using v3.6.10. Before migrating I have created a new MySQL db only for v4.

**Comment by francielelithg** [comment]
> hello :) Did your staging environment was running on an older version before pushing ?If so , could you share the version it was using before?

Hi @Marc-Roig! I was using v3.6.10. Before migrating I have created a new MySQL db only for v4.

**Comment by Marc-Roig** [comment]
ok, so If I understand correctly, in staging you had a fresh db? or the db you have in staging was migrated from the data you had ni 3.6.10.


Just some checks , just in case! https://docs.strapi.io/dev-docs/migration/v3-to-v4/code-migration
If you are migrating from 3.6.10, have you followed the migration docs?

**Comment by francielelithg** [comment]
> ok, so If I understand correctly, in staging you had a fresh db? or the db you have in staging was migrated from the data you had in 3.6.10.
> 
> Just some checks , just in case! https://docs.strapi.io/dev-docs/migration/v3-to-v4/code-migration If you are migrating from 3.6.10, have you followed the migration docs?

Yes, fresh db. Followed migration guides. Thanks for the link, will double check!

**Comment by Boegie19** [comment]
@francielelithg Any updates?

**Comment by jarnix** [comment]
Hi, this issue is happening on a fresh install, just added a collection type and on save, the server restarts and I get this error too:

Error: ER_FK_DUP_NAME: Duplicate FOREIGN KEY constraint name 'blog/admin_permissions_created_by_id_fk'
